Src/Core/WeaponController.cs

```diff
@@ -55,44 +51,32 @@ void Update()
                 swingAudioSource.Play();
             }
 
-            if(!swingBack) attackAngle = 3f * (weaponRenderer.flipY ? -1 : 1);
-            else attackAngle = -3f * (weaponRenderer.flipY ? -1 : 1);
+            attackAngle = swingBack ? -3f * (weaponRenderer.flipY ? -1 : 1) : 3f * (weaponRenderer.flipY ? -1 : 1);
 
             angle -= attackAngle;
 
             if(weaponRenderer.flipY){
-                if(angle >= prevAttackAngle + weapon.swift) swingBack = true;
-
-                if(angle <= prevAttackAngle){
-                    swingBack = false;
-                    attacking = false;
-                }
+                swingBack = (angle >= prevAttackAngle + weapon.swift);
+                swingBack = attacking = (angle <= prevAttackAngle)
             } else {
-                if(angle <= prevAttackAngle - weapon.swift) swingBack = true;
-
-                if(angle >= prevAttackAngle){
-                    swingBack = false;
-                    attacking = false;
-                }
+                swingBack = (angle <= prevAttackAngle - weapon.swift);
+                swingBack = attacking = (angle >= prevAttackAngle);
             }
         } else {
             if(hits.Count > 0) hits = new List<GameObject>();
 
             trailParticles.Stop();
 
             desiredAngle = angleToMouse(transform.parent);
-            angle = moveToward(angle, desiredAngle, weapon.rotationSpeed);
+            prevAttackAngle = angle = moveToward(angle, desiredAngle, weapon.rotationSpeed);
             
             attackAngle = 0f;
-            prevAttackAngle = angle;
 
             float weapAngle = transform.rotation.eulerAngles.z;
-            if(weapAngle < 0) weapAngle = Mathf.Abs(weapAngle) + 180f;
-
-            if(weapAngle >= 90 && weapAngle <= 270) weaponRenderer.flipY = true;
-            else weaponRenderer.flipY = false;
+            weapAngle = weapAngle < 0 ? Mathf.Abs(weapAngle) + 180f : weapAngle;
 
-            if(Input.GetMouseButtonDown(0) && Mathf.Abs(angleDist(desiredAngle, angle)) <= weapon.cone) attacking = true;
+            weaponRenderer.flipY = (weapAngle >= 90 && weapAngle <= 270);
```

Comment:
You should check that these variables shouldnt be false if the condition is false, otherwise the code just breaks

Reply:
Line 79 applies too, but as once its true it goes to another loop, idk